src/openforms/analytics_tools/templates/analytics_tools/piwik_pro_tag_manager.html

```diff
@@ -0,0 +1,33 @@
+{% load solo_tags %}
+{% get_solo 'analytics_tools.AnalyticsToolsConfiguration' as analytics_tools_config %}
+{% if analytics_tools_config.is_piwik_pro_tag_manager_enabled %}
+<!-- Start Piwik PRO tag manager head tag -->
+{% comment %} See https://developers.piwik.pro/en/latest/tag_manager/content_security_policy.html#adding-nonce-to-container-code {% endcomment %}
+<script id="piwik-pro-tag-manager-async" type="text/javascript" nonce="{{ request.csp_nonce }}">
+  (function(window, document, dataLayerName, id) {
+  window[dataLayerName]=window[dataLayerName]||[],window[dataLayerName].push({start:(new Date).getTime(),event:"stg.start"});
+  var scripts=document.getElementsByTagName('script')[0],tags=document.createElement('script');
+  function stgCreateCookie(a,b,c){var d="";if(c){var e=new Date;e.setTime(e.getTime()+24*c*60*60*1e3),d=";expires="+e.toUTCString()}document.cookie=a+"="+b+d+"; path=/"}
+  var isStgDebug=(window.location.href.match("stg_debug")||document.cookie.match("stg_debug"))&&!window.location.href.match("stg_disable_debug");
+  stgCreateCookie("stg_debug",isStgDebug?1:"",isStgDebug?14:-1);
+  var qP=[];dataLayerName!=="dataLayer"&&qP.push("data_layer_name="+dataLayerName),tags.nonce="{{ request.csp_nonce }}",isStgDebug&&qP.push("stg_debug");
+  var qPString=qP.length>0?("?"+qP.join("&")):"";
+  tags.async=!0,tags.src="{{ analytics_tools_config.piwik_pro_url|escapejs }}/"+id+".js"+qPString,
+  scripts.parentNode.insertBefore(tags,scripts);
+  !function(a,n,i){a[n]=a[n]||{};for(var c=0;c<i.length;c++)!function(i){a[n][i]=a[n][i]||{},a[n][i].api=a[n][i].api||function(){
+  var a=[].slice.call(arguments,0);"string"==typeof a[0]&&window[dataLayerName].push({event:n+"."+i+":"+a[0],parameters:[].slice.call(arguments,1)})}}(i[c])}(window,"ppms",["tm","cm"]);
+  })(window, document, 'dataLayer', '{{ analytics_tools_config.piwik_pro_site_id|escapejs }}');
+</script>
+
+<script id="piwik-pro-tag-manager-sync" type="text/javascript" nonce="{{ request.csp_nonce }}">
+  (function(window, document, dataLayerName, id) {
+  function stgCreateCookie(a,b,c){var d="";if(c){var e=new Date;e.setTime(e.getTime()+24*c*60*60*1e3),d=";expires="+e.toUTCString()}document.cookie=a+"="+b+d+"; path=/"}
+  var isStgDebug=(window.location.href.match("stg_debug")||document.cookie.match("stg_debug"))&&!window.location.href.match("stg_disable_debug");
+  stgCreateCookie("stg_debug",isStgDebug?1:"",isStgDebug?14:-1);
+  var qP=[];dataLayerName!=="dataLayer"&&qP.push("data_layer_name="+dataLayerName),isStgDebug&&qP.push("stg_debug");
+  var qPString=qP.length>0?("?"+qP.join("&")):"";
+  document.write('<script src="{{ analytics_tools_config.piwik_pro_url|escapejs }}/'+id+'.sync.js' + qPString + '" nonce="{{ request.csp_nonce }}"></' + 'script>');
```

Comment:
```suggestion
  document.write('<script src="{{ analytics_tools_config.piwik_pro_url|escapejs }}/'+id+'.sync.js' + qPString + '" nonce="{{ request.csp_nonce|escapejs }}"></' + 'script>');
```
?

Reply:
Other script templates don't have it :shrug:

Reply:
`csp_nonce` is not an untrusted value, and docs give the same example: https://django-csp.readthedocs.io/en/latest/nonce.html#middleware

Reply:
it's a very different thing if the variable goes into an HTML tag attribute, for which Django applies regular HTML escaping, versus if you are using a variable inside of Javascript code which requires a different escape context.

Which other script templates don't have it, so that I can take a look at whether they should have it or not?

The thing is that we don't really know what character set is in the CSP nonce - now I happen to know it should be base64 chars so it is *unlikely* that it will cause problems, but the escaping should still be applied if only to serve as a not-bad example of how to use template variables inside of dynamic JS code. Someone else looking at this code could deduce that this is the way to use context variables in JS code inside django templates.

But, if the csp nonce value were to have a <kbd>`</kbd> character, you'll be in trouble because now your Javascript code is broken since a literal `'` is rendered.

Reply:
Yes sorry I thought you commented on the csp nonce tag here:
https://github.com/open-formulieren/open-forms/blob/7c89d2746199e8a4b6c466fbc6bec37e0a6fb315/src/openforms/analytics_tools/templates/analytics_tools/piwik_pro_tag_manager.html#L6

So in this case yes it makes sense!